Header header
ObjectInfo[] object_infos
